pkg/daemon/ceph/osd/daemon.go

```diff
@@ -415,6 +415,18 @@ func getAvailableDevices(context *clusterd.Context, agent *OsdAgent) (*DeviceOsd
 				}
 				matchedDevice = desiredDevice
 
+				if matchedDevice.DeviceClass == "" {
```

Comment:
For the PVC use case, I think we should read the content of the annotation if specified, if not follow the current logic. See how it's done in the provisioning https://github.com/rook/rook/blob/3b10a21d5dcfafd424eddb78a5cba89cda634ecb/pkg/daemon/ceph/osd/volume.go#L358-L361